lib/galaxy/tools/parser/output_actions.py

```diff
@@ -243,7 +243,10 @@ def get_value(self, other_values):
                 # if this is an HDCA for instance let reverse.ext grab
                 # the reverse element and then continue for loop to grab
                 # dataset extension
-                value = value.collection[attr_name].element_object
+                try:
+                    value = value.collection[attr_name].element_object
+                except KeyError:
+                    value = value.child_collection[attr_name].element_object
             elif hasattr(value, "collection") and value in COLLECTION_ATTRIBUTES:
                 value = getattr(value.collection, attr_name)
             else:
```

Comment:
If that above fix is doing what I think, I suppose we should have:

```
elif hasattr(value, "child_collection") and value in COLLECTION_ATTRIBUTES:
    value = getattr(value.collection, attr_name)
```

here as well - or no? I can follow up in a separate PR and find testable a path to the logic.

Reply:
That makes sense, except I can't figure out in which situation you can get `value` to have a collection attribute and at the same time to be equal to `collection_type` ?

Reply:
I have no clue, that looks super speculative in retrospect. Lets just drop this ``COLLECTION_ATTRIBUTES`` non-sense in dev and see if any tests break but merge this as is so it can be backported to 18.01/18.05?